### PR TITLE
Add forward and reply icons on threads

### PR DESCRIPTION
**Pull request by TommyDL-Infomaniak**

### CODE REVIEW

File: app/src/main/res/layout/cardview_thread_item.xml [hunk 1]
```diff
@@ -213,6 +213,40 @@
                     app:layout_constraintTop_toTopOf="@id/expeditor"
                     tools:text="@tools:sample/date/hhmm" />
 
+                <FrameLayout
+                    android:id="@+id/iconReplyOrForwardLayout"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="start"
```
Comment (LunarX) [inline app/src/main/res/layout/cardview_thread_item.xml:220]: You should not need this layout_gravity

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 2]
```diff
@@ -218,6 +218,15 @@ class ThreadListAdapter @Inject constructor(
...
             draftPrefix.isVisible = hasDrafts
 
+            iconReply.isGone = true
+            iconForward.isGone = true
+
+            if (thread.isAnswered) {
+                iconReply.isVisible = true
+            } else if (thread.isForwarded) {
+                iconForward.isVisible = true
+            }
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:228]: Make sure how priority should truly be computed

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt (hunk 2 above)
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:228]: Don't assign it twice.
Assign it everytime for the 2 values, to the correct value.

File: app/src/main/res/drawable/ic_forward_arrow.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<!--
```
Comment (KevinBoulongne) [inline app/src/main/res/drawable/ic_forward_arrow.xml:1]:
```suggestion
<?xml version="1.0" encoding="utf-8"?><!--
```

File: app/src/main/res/drawable/ic_reply_arrow.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<!--
```
Comment (KevinBoulongne) [inline app/src/main/res/drawable/ic_reply_arrow.xml:1]:
```suggestion
<?xml version="1.0" encoding="utf-8"?><!--
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt (hunk 2 above)
Comment (TommyDL-Infomaniak) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt]: The priority is given to the last action on a message (replied or forwarded)

**Review by tevincent — APPROVED** [review]

File: app/src/main/res/drawable/ic_forward_arrow.xml
Comment (LunarX) [inline app/src/main/res/drawable/ic_forward_arrow.xml:1]: Ar there two icons the same as their outlined version but filled? If so, maybe we could try and find a common name between the two, except this version adds `_filled` at the end of its name?

File: app/src/main/res/layout/cardview_thread_item.xml [hunk 5]
```diff
@@ -213,6 +213,39 @@
...
 
+                <FrameLayout
+                    android:id="@+id/iconReplyOrForwardLayout"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    app:layout_constraintBottom_toBottomOf="@id/mailSubject"
+                    app:layout_constraintEnd_toStartOf="@+id/mailSubject"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="@+id/mailSubject"
+                    tools:visibility="visible">
```
Comment (LunarX) [inline app/src/main/res/layout/cardview_thread_item.xml:224]: Not sur this `tools:visibility` is gonna be of much help here

**Review by LunarX — CHANGES_REQUESTED** [review]

**Review by LunarX — CHANGES_REQUESTED** [review]
We don't do anything for people who already had badly computed isAnswered and isForwarded values inside their Thread objects already stored on realm?

File: app/src/main/res/drawable/ic_forward_arrow.xml
Comment (TommyDL-Infomaniak) [reply app/src/main/res/drawable/ic_forward_arrow.xml:1]: They are, I have made the change.

**Comment by TommyDL-Infomaniak** [comment]
> We don't do anything for people who already had badly computed isAnswered and isForwarded values inside their Thread objects already stored on realm?

After discussion with the iOS team, it was decided that it is not worth it.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 6]
```diff
@@ -209,6 +209,18 @@ class ThreadListAdapter @Inject constructor(
...
+        // could be both answered and forwarded (for new threads, this is impossible).
+        fun computeReplyAndForwardIcon(isAnswered: Boolean, isForwarded: Boolean): Pair<Boolean, Boolean> {
+            return if (isAnswered) {
+                Pair(true, false)
+            } else if (isForwarded) {
+                Pair(false, true)
+            } else {
+                Pair(false, false)
+            }
+        }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:222]:
```suggestion
    fun computeReplyAndForwardIcon(isAnswered: Boolean, isForwarded: Boolean): Pair<Boolean, Boolean> {
        return when {
            isAnswered -> true to false
            isForwarded -> false to true
            else -> false to false
        }
    }
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/res/drawable/ic_forward_arrow.xml
Comment (LunarX) [reply app/src/main/res/drawable/ic_forward_arrow.xml:1]: The icon `ic_email_action_reply` is not named like `ic_reply_arrow_filled`

**Review by LunarX — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 7]
```diff
@@ -209,6 +209,16 @@ class ThreadListAdapter @Inject constructor(
 
         displayFolderName(thread)
 
+        // This method is only useful for old threads already stored in Realm, where they
+        // could be both answered and forwarded (for new threads, this is impossible).
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:213]:
```suggestion
        // This method is only useful for old Threads already stored in Realm, where they
        // could be both answered and forwarded (for new Threads, this is impossible).
```

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 8]
```diff
@@ -217,6 +227,11 @@ class ThreadListAdapter @Inject constructor(
 
             draftPrefix.isVisible = hasDrafts
 
+            computeReplyAndForwardIcon(thread.isAnswered, thread.isForwarded).let { (iconReplyIsVisible, iconForwardIsVisible) ->
+                iconReply.isVisible = iconReplyIsVisible
+                iconForward.isVisible = iconForwardIsVisible
+            }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:233]:
```suggestion
            val (isIconReplyVisible, isIconForwardVisible) = computeReplyAndForwardIcon(thread.isAnswered, thread.isForwarded)
            iconReply.isVisible = isIconReplyVisible
            iconForward.isVisible = isIconForwardVisible
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**